### PR TITLE
tests: net: ipv6_fragment: Improve IPv6 fragment testing

### LINKED ISSUE #51788: Questionable test code in ipv6_fragment test
**Describe the bug**
Some of the code in the ipv6_fragment test does not seem to be valid. The function `udp_data_received` looks like it controls processing of received IPv6 fragment data, however the function is never called. The actual test cases in the file are not completely clear, from what the layout looks like, data should be sent from one interface to another, however this never happens, in fact despite the second interface being setup, it is actually completely unused in the whole test, you can remove the whole interface from the test and run it and the test still passes.
I therefore think that this test case is incomplete, and it certainly does not test IPv6 reassembly for the higher network layers.

**Expected behavior**
All IPv6 fragment functionality should be tested, including TCP/UDP packets and from one network device to another.

**Impact**
Test case is flawed

**Environment (please complete the following information):**
 - Commit SHA or Version used: b219b5cba721b43a2c4fe71f5845f0abc852e7f6

### PR DESCRIPTION
**Pull request by rlubos**
Cleanup and improve the `ipv6_fragment` test suite
* Remove unused dummy interface
* Fix `wait_data` semaphore usage (it should only be given after all expected fragments are processed)
* Add a new test case, to verify fragmentation and reassembly in the loopback scenario
* Improve `test_recv_ipv6_fragment` test case to actually verify the reassembled packet.

Fixes #51788

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]
LGTM

File: tests/net/ipv6_fragment/src/main.c [hunk 1]
```diff
@@ -1323,11 +1379,181 @@ static int verify_fragment(struct net_pkt *pkt)
...
+
+	NET_DBG("Data %p received", pkt);
+
+	zassert_equal(test_type, IPV6_UDP_LOOPBACK,
+		      "Packet should only be received in UDP loopback test");
+	zassert_equal(net_pkt_get_len(pkt), expected_pkt_length, "Invalid packet length");
+
+	/* Verify IPv6 header is valid */
+	zassert_equal(hdr->vtc, ipv6_udp[offsetof(struct net_ipv6_hdr, vtc)],
+		      "IPv6 header vhl mismatch");
```
Comment (nordicjm) [inline tests/net/ipv6_fragment/src/main.c:1510]: vtc?

File: tests/net/ipv6_fragment/src/main.c [hunk 2]
```diff
@@ -1951,23 +2216,99 @@ static uint8_t ipv6_reass_frag2[] = {
...
+	uint16_t i;
+	uint8_t expected_data = 0;
+
+	NET_DBG("Data %p received", pkt);
+
+	zassert_equal(net_pkt_get_len(pkt), expected_pkt_length, "Invalid packet length");
+
+	/* Verify IPv6 header is valid */
+	zassert_equal(hdr->vtc, ipv6_reass_frag1[offsetof(struct net_ipv6_hdr, vtc)],
+		      "IPv6 header vhl mismatch");
```
Comment (nordicjm) [inline tests/net/ipv6_fragment/src/main.c:2242]: vtc?

File: tests/net/ipv6_fragment/src/main.c (hunk 1 above)
Comment (rlubos) [reply tests/net/ipv6_fragment/src/main.c]: Should be fixed now.

**Review by nordicjm — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**